python/lsst/daf/butler/core/storageClassDelegate.py

```diff
@@ -396,3 +399,38 @@ def selectResponsibleComponent(cls, derivedComponent: str, fromComponents: Set[O
             from the supplied options.
         """
         raise NotImplementedError("This delegate does not support derived components")
+
+    def copy(self, inMemoryDataset: Any) -> Any:
+        """Copy the supplied python type and return the copy.
+
+        Parameters
+        ----------
+        inMemoryDataset : `object`
+            Object to copy.
+
+        Returns
+        -------
+        copied : `object`
+            A copy of the supplied object. Can be the same object if the
+            object is known to be read-only.
+
+        Raises
+        ------
+        NotImplementedError
+            Raised if none of the default methods for copying work.
+
+        Notes
+        -----
+        The default implementation uses `copy.deepcopy()`.
+        It is generally expected that this method is the equivalent of a deep
+        copy. Subclasses can override this method if they already know the
+        optimal approach for deep copying.
+        """
+
+        try:
+            return copy.deepcopy(inMemoryDataset)
+        except Exception as e:
+            raise NotImplementedError(
```

Comment:
Can we add a test to exercise this failure?

Reply:
I meant to do that but I utterly failed to think of a class that failed. Will think again.